pos_print_sale_order/static/src/js/SaleOrderManagementScreen.esm.js

```diff
@@ -0,0 +1,94 @@
+                ) {
+                    const printActions =
+                        this.env.pos.config.print_sales_order_ids.filter((id) => {
+                            return this.env.pos.ir_actions_report.some(
+                                (report) => report.id === id
+                            );
+                        });
+
+                    // Show a selection popup for the user to choose what to print
+                    const {confirmed: popupConfirmed, payload: popupSelectedOption} =
+                        await this.showPopup("SelectionPopup", {
+                            title: this.env._t("What do you want to print?"),
+                            list: printActions.map((actionId) => ({
+                                id: actionId,
+                                item: actionId,
+                                label: this.env.pos.ir_actions_report.find(
+                                    (report) => report.id === actionId
+                                ).name,
+                            })),
+                        });
+
+                    if (popupConfirmed) {
+                        try {
+                            await this.env.legacyActionManager.do_action(
+                                popupSelectedOption,
+                                {
+                                    additional_context: {
+                                        active_ids: [this.clickedOrder.id],
+                                    },
+                                }
+                            );
+                        } catch (error) {
+                            if (error instanceof Error) {
+                                throw error;
+                            } else {
+                                this.showPopup("ErrorPopup", {
+                                    title: this.env._t("Network Error"),
+                                    body: this.env._t("Unable to download the report."),
+                                });
+                            }
+                        }
+                    }
+                } else {
+                    this.showPopup("ErrorPopup", {
+                        title: this.env._t("Print Error"),
+                        body: this.env._t(
+                            "Please choose which sale orders to print in the POS sales configuration."
+                        ),
+                    });
+                }
+            }
+            return {confirmed, payload};
+        }
+        showPopup(name, props) {
+            if (name == "SelectionPopup" && this.additionalSelectionList) {
+                props.list = [...props.list, ...this.additionalSelectionList];
+            }
+            return super.showPopup(name, props).then(({confirmed, payload}) => {
+                // Handle printing based on user selections
+                const result = this._printSaleOrder(confirmed, payload);
```

Comment:
It seems to me that there is not enough check for the current popup, otherwise this callback will be called for any popup.